redisson-tomcat/redisson-tomcat-8/src/main/java/org/redisson/tomcat/RedissonSessionManager.java

```diff
@@ -67,7 +67,9 @@
 
     private final String nodeId = UUID.randomUUID().toString();
 
-    private static ValveBase updateValve;
+    // For the test, by placing this in a static there is only 1 update valve, and its registered to the first
+    // Tomcat instance, so that server2 never triggers the update valve, as they both have their own Engine/pipeline instances
+    private ValveBase updateValve;
```

Comment:
please look at this issue https://github.com/redisson/redisson/issues/2178

Reply:
OK, we may not be able to commit this change as is. but without it server2 in the tests will never have the valves.

Would it be better to instead of holding a reference to the valve(s) directly to lookup in `getEngine().getPipeline().getValves()` and loop through the list finding the ones which are instances of `ValveBase` to stop/remove them at the end.

Reply:
Added a new commit showing how this could work, removing the field(s) all together